### PR TITLE
DOC-3712 Issue TA 82309

### PR DESCRIPTION
**Pull request by andf-crl**
Addresses: DOC-3712

- Issues new TA 82309, affecting changefeeds in v22.1.0

[advisories/a82309.md](https://deploy-preview-14012--cockroachdb-docs.netlify.app/docs/advisories/a82309)

### CODE REVIEW

**Review by shermanCRL — APPROVED** [review]

File: advisories/a82309.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
+---
+title: Technical Advisory 82309
+advisory: A-82309
+summary: During or after an upgrade from CockroachDB v21.2.x to v22.1.0, existing changefeeds will stop emitting data.
+toc: true
+affected_versions: v22.1.0-alpha.1-v22.1.0
```
Comment (amruss) [inline advisories/a82309.md:6]: should this be a comma between versions?

Comment (andf-crl) [reply advisories/a82309.md:6]: This field shows up verbatim in the `Affected Versions` column here: https://deploy-preview-14012--cockroachdb-docs.netlify.app/docs/advisories/index.html

**Review by amruss — APPROVED** [review]
I added one NIT comment but besides that it LGTM

**Review by taroface — APPROVED** [review]
I have some (mostly) formatting and wording suggestions, but overall LGTM!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/14012#-:-N3aXq7MCb_TG1LdJ_h5:b-akxf3v)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @amruss, @andf-crl, and @miretskiy)

___
*[`advisories/a82309.md` line 15 at r2](https://reviewable.io/reviews/cockroachdb/docs/14012#-N3aUEbt2v8apiIVuo_x:-N3aUEbt2v8apiIVuo_y:b-bt9xkq) ([raw file](https://github.com/cockroachdb/docs/blob/886313451867eeb5e8b9736899d3dad53d7601b3/advisories/a82309.md#L15)):*
> ```Markdown
> ## Description
> 
> During or after an upgrade from CockroachDB v21.2.x to v22.1.0, existing changefeeds will stop emitting data.
> ```

Suggestion: Add link to change-data-capture-overview.html



_Code quote:_
```Markdown
changefeeds
```
___
*[`advisories/a82309.md` line 23 at r2](https://reviewable.io/reviews/cockroachdb/docs/14012#-N3aU_Gs0ZumFHzr-RgK:-N3aU_Gt6jOSs9Q5lYKk:birbgbe) ([raw file](https://github.com/cockroachdb/docs/blob/886313451867eeb5e8b9736899d3dad53d7601b3/advisories/a82309.md#L23)):*
> ```Markdown
> Cockroach Labs has discovered a bug that results in changefeeds failing to emit data during or after an upgrade to v22.1.0 from v21.2.x. This failure may be relatively silent, in that no errors will appear in the logs.
> 
> Users are advised to delay upgrades until the upcoming v22.1.1 patch release, targeted for June 6.
> ```

FYI the docs PR is now open, so in case this TA doesn't get merged until Monday, the release notes should be ready to link here.
___
*[`advisories/a82309.md` line 29 at r2](https://reviewable.io/reviews/cockroachdb/docs/14012#-N3aVxqqEbvu2Lxtfvqo:-N3aVxqqEbvu2Lxtfvqp:b-k7zlvx) ([raw file](https://github.com/cockroachdb/docs/blob/886313451867eeb5e8b9736899d3dad53d7601b3/advisories/a82309.md#L29)):*
> ```Markdown
> - You observe that your downstream [sink](../{{site.versions["stable"]}}/changefeed-sinks.html) (such as Kafka) is not receiving data.
> - After the upgrade begins, metrics related to the number of messages (`changefeed.emitted_messages`) and message sizes (`changefeed.emitted_bytes`) will be close to `0`. This metric will begin dropping immediately, as soon as nodes start upgrading.
> - The decay in the number of emitted messages may be gradual (depending on the upgrade speed), but once all nodes upgrade, `0` messages will be emitted.
> ```

Nit: for clarity, should this be either "all nodes have upgraded" or "all nodes have begun upgrading"? Not sure which is correct.



_Suggestion:_
```Markdown
all nodes upgrade
```
___
*[`advisories/a82309.md` line 31 at r2](https://reviewable.io/reviews/cockroachdb/docs/14012#-N3aW7tyC1KvXqWV5nf_:-N3aW7tyC1KvXqWV5nfa:b-3zbpgl) ([raw file](https://github.com/cockroachdb/docs/blob/886313451867eeb5e8b9736899d3dad53d7601b3/advisories/a82309.md#L31)):*
> ```Markdown
> - The decay in the number of emitted messages may be gradual (depending on the upgrade speed), but once all nodes upgrade, `0` messages will be emitted.
> 
> If the changefeeds run with the [`resolved` option](../{{site.versions["stable"]}}/create-changefeed.html#resolved-option), your sink will continue receiving “RESOLVED” messages but will not receive any data messages. Changefeeds will *appear* to be up to date, however these "RESOLVED" messages are incorrect and should be disregarded.
> ```

Should this rather be in code formatting?



_Code quote:_
```Markdown
“RESOLVED”
```
___
*[`advisories/a82309.md` line 31 at r2](https://reviewable.io/reviews/cockroachdb/docs/14012#-N3aWADA6WumN4V2dLeX:-N3aWADA6WumN4V2dLeY:b-y494kh) ([raw file](https://github.com/cockroachdb/docs/blob/886313451867eeb5e8b9736899d3dad53d7601b3/advisories/a82309.md#L31)):*
> ```Markdown
> - The decay in the number of emitted messages may be gradual (depending on the upgrade speed), but once all nodes upgrade, `0` messages will be emitted.
> 
> If the changefeeds run with the [`resolved` option](../{{site.versions["stable"]}}/create-changefeed.html#resolved-option), your sink will continue receiving “RESOLVED” messages but will not receive any data messages. Changefeeds will *appear* to be up to date, however these "RESOLVED" messages are incorrect and should be disregarded.
> ```

Per our style, this should be bolded instead.



_Code quote:_
```Markdown
*appear*
```
___
*[`advisories/a82309.md` line 31 at r2](https://reviewable.io/reviews/cockroachdb/docs/14012#-N3aWDFx33c_3rOMcmhk:-N3aWDFx33c_3rOMcmhl:b-89a4l7) ([raw file](https://github.com/cockroachdb/docs/blob/886313451867eeb5e8b9736899d3dad53d7601b3/advisories/a82309.md#L31)):*
> ```Markdown
> - The decay in the number of emitted messages may be gradual (depending on the upgrade speed), but once all nodes upgrade, `0` messages will be emitted.
> 
> If the changefeeds run with the [`resolved` option](../{{site.versions["stable"]}}/create-changefeed.html#resolved-option), your sink will continue receiving “RESOLVED” messages but will not receive any data messages. Changefeeds will *appear* to be up to date, however these "RESOLVED" messages are incorrect and should be disregarded.
> ```

Ditto



_Code quote:_
```Markdown
"RESOLVED"
```
___
*[`advisories/a82309.md` line 31 at r2](https://reviewable.io/reviews/cockroachdb/docs/14012#-N3aWMAX-dY5jTaiAIjv:-N3aWMAX-dY5jTaiAIjw:undefined) ([raw file](https://github.com/cockroachdb/docs/blob/886313451867eeb5e8b9736899d3dad53d7601b3/advisories/a82309.md#L31)):*
> ```Markdown
> - The decay in the number of emitted messages may be gradual (depending on the upgrade speed), but once all nodes upgrade, `0` messages will be emitted.
> 
> If the changefeeds run with the [`resolved` option](../{{site.versions["stable"]}}/create-changefeed.html#resolved-option), your sink will continue receiving “RESOLVED” messages but will not receive any data messages. Changefeeds will *appear* to be up to date, however these "RESOLVED" messages are incorrect and should be disregarded.
> ```





_Suggestion:_
```Markdown
Although changefeeds will **appear** to be up to date, these `RESOLVED` messages are incorrect and should be disregarded.
```
___
*[`advisories/a82309.md` line 33 at r2](https://reviewable.io/reviews/cockroachdb/docs/14012#-N3aWUxQDU3r3PHSifgU:-N3aWUxQDU3r3PHSifgV:undefined) ([raw file](https://github.com/cockroachdb/docs/blob/886313451867eeb5e8b9736899d3dad53d7601b3/advisories/a82309.md#L33)):*
> ```Markdown
> If the changefeeds run with the [`resolved` option](../{{site.versions["stable"]}}/create-changefeed.html#resolved-option), your sink will continue receiving “RESOLVED” messages but will not receive any data messages. Changefeeds will *appear* to be up to date, however these "RESOLVED" messages are incorrect and should be disregarded.
> 
> This is resolved in CockroachDB by [PR #82312](https://github.com/cockroachdb/cockroach/pull/82312).
> ```





_Suggestion:_
```Markdown
PR [#82312]
```
___
*[`advisories/a82309.md` line 35 at r2](https://reviewable.io/reviews/cockroachdb/docs/14012#-N3aWZgFA3S-tAtiOlrU:-N3aWZgFA3S-tAtiOlrV:undefined) ([raw file](https://github.com/cockroachdb/docs/blob/886313451867eeb5e8b9736899d3dad53d7601b3/advisories/a82309.md#L35)):*
> ```Markdown
> This is resolved in CockroachDB by [PR #82312](https://github.com/cockroachdb/cockroach/pull/82312).
> 
> The fix will be applied to the upcoming maintenance version [v22.1.1](../releases/v22.1.html) of CockroachDB, currently scheduled for June 6th, 2022.
> ```





_Suggestion:_
```Markdown
6
```
___
*[`advisories/a82309.md` line 45 at r2](https://reviewable.io/reviews/cockroachdb/docs/14012#-N3aWk0Q3G1PtQJBG2PP:-N3aWk0Q3G1PtQJBG2PQ:undefined) ([raw file](https://github.com/cockroachdb/docs/blob/886313451867eeb5e8b9736899d3dad53d7601b3/advisories/a82309.md#L45)):*
> ```Markdown
> Avoid creation of changefeeds during the upgrade process, i.e., while in a mixed-version state. If you have created changefeeds before or during the upgrade process, you should expect them to experience the above-described issue.
> 
> If you have already upgraded, and observed a failure of changefeeds, you will need to [cancel](../{{site.versions["stable"]}}/create-changefeed.html#manage-a-changefeed) and recreate your changefeed(s).
> ```





_Suggestion:_
```Markdown
upgraded to v22.1
```
___
*[`advisories/a82309.md` line 49 at r2](https://reviewable.io/reviews/cockroachdb/docs/14012#-N3aX3l03yE_wCpXyrAf:-N3aX3l03yE_wCpXyrAg:bk6g7u4) ([raw file](https://github.com/cockroachdb/docs/blob/886313451867eeb5e8b9736899d3dad53d7601b3/advisories/a82309.md#L49)):*
> ```Markdown
> You can make a choice as to how much “catch up” data you wish to emit from the newly-created changefeeds:
> 
> In the default case, creating a changefeed will emit (scan) the entire table to your sink, and then begin emitting all new events. This does not require any special options, as it is the default.
> ```

I suggest bulleting the two options (and indenting their respective contents under the bullets, if that isn't too complicated), for visual clarity.
___
*[`advisories/a82309.md` line 54 at r2](https://reviewable.io/reviews/cockroachdb/docs/14012#-N3aWsol3sS7uZ-3k9YE:-N3aWsol3sS7uZ-3k9YF:b-t9thbi) ([raw file](https://github.com/cockroachdb/docs/blob/886313451867eeb5e8b9736899d3dad53d7601b3/advisories/a82309.md#L54)):*
> ```Markdown
> ~~~ sql
> CREATE CHANGEFEED FOR TABLE name, name2, name3 
> INTO '<sink URI>'
> ```

Nit: Maybe use the `{ }` brackets in this syntax?



_Code quote:_
```Markdown
<
```
___
*[`advisories/a82309.md` line 58 at r2](https://reviewable.io/reviews/cockroachdb/docs/14012#-N3aWzWA2CvPiXq1kGWR:-N3aWzWA2CvPiXq1kGWS:undefined) ([raw file](https://github.com/cockroachdb/docs/blob/886313451867eeb5e8b9736899d3dad53d7601b3/advisories/a82309.md#L58)):*
> ```Markdown
> ~~~
> 
> You may wish to be more precise in restarting (catching up) from the point in time where the failure began. This is likely preferable for large tables where you don’t wish to incur the full initial scan. To do this, you will need to choose a point in time. We recommend choosing a few minutes before your upgrade began.
> ```





_Suggestion:_
```Markdown
do not
```
___
*[`advisories/a82309.md` line 58 at r2](https://reviewable.io/reviews/cockroachdb/docs/14012#-N3aX8eMEcg1PPNvKocL:-N3aX8eMEcg1PPNvKocM:b3a3zfc) ([raw file](https://github.com/cockroachdb/docs/blob/886313451867eeb5e8b9736899d3dad53d7601b3/advisories/a82309.md#L58)):*
> ```Markdown
> ~~~
> 
> You may wish to be more precise in restarting (catching up) from the point in time where the failure began. This is likely preferable for large tables where you don’t wish to incur the full initial scan. To do this, you will need to choose a point in time. We recommend choosing a few minutes before your upgrade began.
> ```

Ditto re: bulleting the option
___
*[`advisories/a82309.md` line 60 at r2](https://reviewable.io/reviews/cockroachdb/docs/14012#-N3aXEZ5BssTlqMpfwNm:-N3aXEZ5BssTlqMpfwNn:bjuk6df) ([raw file](https://github.com/cockroachdb/docs/blob/886313451867eeb5e8b9736899d3dad53d7601b3/advisories/a82309.md#L60)):*
> ```Markdown
> You may wish to be more precise in restarting (catching up) from the point in time where the failure began. This is likely preferable for large tables where you don’t wish to incur the full initial scan. To do this, you will need to choose a point in time. We recommend choosing a few minutes before your upgrade began.
> 
> **Note:** The usual method of [retrieving the high-water timestamp cursor](../{{site.versions["stable"]}}/create-changefeed.html#start-a-new-changefeed-where-another-ended) is **invalid** in the presence of this bug.
> ```

Instead of using **Note:** maybe use the `info` callout?
___
*[`advisories/a82309.md` line 65 at r2](https://reviewable.io/reviews/cockroachdb/docs/14012#-N3aXJDw21W61APiZlzm:-N3aXJDw21W61APiZlzn:bmizq7) ([raw file](https://github.com/cockroachdb/docs/blob/886313451867eeb5e8b9736899d3dad53d7601b3/advisories/a82309.md#L65)):*
> ```Markdown
> ~~~ sql
> CREATE CHANGEFEED FOR TABLE name, name2, name3 
> INTO '<sink URI>'
> ```

Ditto re: brackets



_Code quote:_
```Markdown
<
```
___
*[`advisories/a82309.md` line 85 at r2](https://reviewable.io/reviews/cockroachdb/docs/14012#-N3aXPW--KiOjG8c-jLC:-N3aXPW--KiOjG8c-jLD:bopfwf1) ([raw file](https://github.com/cockroachdb/docs/blob/886313451867eeb5e8b9736899d3dad53d7601b3/advisories/a82309.md#L85)):*
> ```Markdown
>     ~~~
> 
> Alternatively, you may wish to restart changefeeds to pick up new changes starting *now*, skipping the initial scan, and skipping change events since the upgrade began. You can do this with the following:
> ```

Not sure if bold is appropriate here... I defer to you



_Suggestion:_
```Markdown
**now**
```
___
*[`advisories/a82309.md` line 90 at r2](https://reviewable.io/reviews/cockroachdb/docs/14012#-N3aXVLi8N2XNxDGSKn6:-N3aXVLi8N2XNxDGSKn7:bmizq7) ([raw file](https://github.com/cockroachdb/docs/blob/886313451867eeb5e8b9736899d3dad53d7601b3/advisories/a82309.md#L90)):*
> ```Markdown
> ~~~ sql
> CREATE CHANGEFEED FOR TABLE name, name2, name3 
> INTO '<sink URI>'
> ```

Ditto re: brackets



_Code quote:_
```Markdown
<
```
___
*[`advisories/a82309.md` line 98 at r2](https://reviewable.io/reviews/cockroachdb/docs/14012#-N3aXYo1-MhhTufTMi2F:-N3aXYo1-MhhTufTMi2G:undefined) ([raw file](https://github.com/cockroachdb/docs/blob/886313451867eeb5e8b9736899d3dad53d7601b3/advisories/a82309.md#L98)):*
> ```Markdown
> During and after the upgrade process from v21.2.x to v22.1.0, changefeeds will stop emitting data. This failure may go undetected, depending on your monitoring.
> 
> Please reach out to the [support team](https://support.cockroachlabs.com/) if more information or assistance is needed.
> ```





_Suggestion:_
```Markdown
you need more information or assistance
```

File: advisories/a82309.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+This public issue is tracked by [#82309](https://github.com/cockroachdb/cockroach/issues/82309).
+
+## Mitigation
+
+If you have not yet upgraded to a v22.1 major version, wait until the upcoming v22.1.1 patch release or a later release.
+
+Avoid creation of changefeeds during the upgrade process, i.e., while in a mixed-version state. If you have created changefeeds before or during the upgrade process, you should expect them to experience the above-described issue.
+
+If you have already upgraded, and observed a failure of changefeeds, you will need to [cancel](../{{site.versions["stable"]}}/create-changefeed.html#manage-a-changefeed) and recreate your changefeed(s).
```
Comment (shermanCRL) [inline advisories/a82309.md:45]: Let’s bold “cancel and recreate”, this is the most important point.

**Review by andf-crl — COMMENTED** [review]
TYFRs, Ryan I've made all your changes. TA to release today per engineering, but I will update Monday with link to v22.1.1 anchor location.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/14012#-:-N3dmVEWA-oi0x8m1fHV:bdmf70a)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @amruss, @andf-crl, and @miretskiy)

**Pull request merged**